### PR TITLE
[maintenance] prepare next release for sklearn 1.8 API breaking changes

### PR DESCRIPTION
**Pull request by icfaust**
## Description

This PR relies on issues occurring in the Nightly azure pipelines run.
This cover three main issues:

- Proper support of ```ensure_all_finite``` in daal4py's ```pairwise_distances```
- ```LogisticRegression```'s removal of the ```multi_class``` keyword (scikit-learn/scikit-learn#32073)
- Removal of the ```CalibratedClassifierCV```'s ```cv``` keyword (used in SVM's probability=True calculations)
- Array API support issues with respect to the ```asarray``` method, which is set to fail in some frameworks (dpctl, array_api_strict) depending on the set device

Which includes the following additional changes:
- Removal of unused imports in daal4py.sklearn.logistic
- Adding the missing test_logistic sklearn conformance testing in public CI
- Refactoring ```logistic_regression_path``` to separate sklearn fallback for easy adaption of signature changes
- Deleting unusued logic in daal4py.sklearn.logistic (shown by pyright)
- Deselection of ```IncrementalEmpiricalCovariance.mahalanobis``` when numpy < 2.0 for array API testing with ```array_api_strict```.  This is due to conditions in both ```array_api_strict``` and ```numpy``` which clash (numpy not having true ```__dlpack__``` support, and ```array_api_strict``` denying the ```__array__``` attribute.

The results from the Nightly will not be green, as some of the necessary 1.8 changes have not been integrated into sklearn main yet.  This means that the changes in this PR assume that no major changes will occur in scikit-learn/scikit-learn#32073 before merging.

Errors related to array API PCA were deduced to be caused by issues in scikit-learn and not in our own repository.  A PR has been opened to sklearn to correct their issue (scikit-learn/scikit-learn#32198).

Deselection of tests which fail by numerical issues are not done. This is because it is not a fundamental breaking of the user experience, and should only be integrated when CI is upgraded to sklearn 1.8 support.

In order to show the viability of these changes, Nightly can show some of it, but manual testing with development branches is necessary (for the LogsiticRegression changes).

Follow-up work still needs to be done for mathematical validity in #1879, so this solution is a stop-gap.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

### CODE REVIEW

File: daal4py/sklearn/utils/validation.py [hunk 1]
```diff
@@ -280,9 +280,9 @@ def _daal_check_array(
     array_converted : object
         The converted and validated array.
     """
-    if force_all_finite not in (True, False, "allow-nan"):
+    if force_all_finite not in (True, False, "allow-nan", None):
```
Comment (david-cortes-intel) [inline daal4py/sklearn/utils/validation.py:283]: `None in None` does not yield `True`.

Comment (icfaust) [reply daal4py/sklearn/utils/validation.py:283]: image

Comment (david-cortes-intel) [reply daal4py/sklearn/utils/validation.py:283]: Thanks, seems to work when put on a tuple.

**Comment by icfaust** [comment]
Last nightly run (https://dev.azure.com/daal/daal4py/_build/results?buildId=51046&view=results) shows how many errors are due to LogisticRegression accuracy. These will be deselected.

**Comment by icfaust** [comment]
DO NOT MERGE IN CURRENT STATE: this is validating the sklearn branch which is making changes to the LogisticRegression API (https://dev.azure.com/daal/daal4py/_build/results?buildId=51280&view=results).

**Comment by icfaust** [comment]
This is a run which shows that LogisticRegression API functions correctly. Must be checked in a case by case basis. (https://dev.azure.com/daal/daal4py/_build/results?buildId=51343&view=results)

File: daal4py/mb/__init__.py [hunk 2]
```diff
@@ -54,14 +54,17 @@ def convert_model(model) -> "GBTDAALModel | LogisticDAALModel":
...
         if model.classes_.shape[0] > 2:
-            if (model.multi_class == "ovr") or (
-                model.multi_class == "auto" and model.solver == "liblinear"
+            if hasattr(model, "multi_class") and (
+                (model.multi_class == "ovr")
+                or (model.multi_class == "auto" and model.solver == "liblinear")
             ):
                 raise TypeError(
                     "Supplied 'model' object is a linear classifier, but not multinomial logistic"
                     " (hint: pass multi_class='multinomial' to 'LogisticRegression')."
```
Comment (avolkov-intel) [inline daal4py/mb/__init__.py]: Should we add information about mult_class deprication here?

Comment (icfaust) [reply daal4py/mb/__init__.py]: Sounds good, will leave a note.

File: daal4py/sklearn/linear_model/logistic_path.py [hunk 3]
```diff
@@ -51,27 +45,13 @@
     from sklearn.linear_model._logistic import (
         LogisticRegression as LogisticRegression_original,
     )
-    from sklearn.linear_model._logistic import (
-        _check_multi_class,
-        _check_solver,
-        _fit_liblinear,
-    )
+    from sklearn.linear_model._logistic import _check_solver
```
Comment (avolkov-intel) [inline daal4py/sklearn/linear_model/logistic_path.py]: We can put common imports outside of the if statement

File: tests/test_model_builders.py [hunk 4]
```diff
@@ -2168,15 +2168,21 @@ def test_logreg_builder_sequential_calls():
...
+                LogisticRegression(multi_class="ovr"),
+                3,
+            ),
+            (
+                LogisticRegression(multi_class="multinomial"),
+                2,
+            ),
+        ]
+        if not sklearn_check_version("1.8")
+        else []
```
Comment (Vika-F) [inline tests/test_model_builders.py]: Should it be something like this:
```suggestion
        else [(LogisticRegression(), 2)]
```
?

Comment (icfaust) [reply tests/test_model_builders.py]: Its checking for multi_class settings that have been removed with a forced use of the `auto` option (and therefore default parameters can't be used in the test).  Good question, and in line with @avolkov-intel 's comment, I have added documentation to it.

Comment (icfaust) [reply tests/test_model_builders.py]: I have actually added ```LogisticRegression(solver="liblinear")``` since that should trigger a case where a linear solver is used. Thanks for the suggestion, it will now be testing again.

File: daal4py/sklearn/linear_model/logistic_path.py (hunk 3 above)
Comment (icfaust) [reply daal4py/sklearn/linear_model/logistic_path.py]: done!

File: sklearnex/svm/_common.py [hunk 5]
```diff
@@ -262,12 +311,26 @@ def _fit_proba(self, X, y, sample_weight=None, queue=None):
...
             clf_base.fit(X, y)
-            self.clf_prob = CalibratedClassifierCV(
-                clf_base,
-                ensemble=False,
-                cv="prefit",
-                method="sigmoid",
-            ).fit(X, y)
+
+            # FrozenEstimator not available for sklearn < 1.4
+            if sklearn_check_version("1.6"):
```
Comment (ethanglaser) [inline sklearnex/svm/_common.py]: Should it be checking for 1.5 then?

**Review by ethanglaser — COMMENTED** [review]
logreg changes look good to me. admittedly I do not know what is going on in the SVM part so i'll defer to others on review of that

There is one public CI fail related to __array__ functionality which is modified in this code, but its also a test that fails often in public CI - do the error messages look real?

File: sklearnex/svm/_common.py (hunk 5 above)
Comment (icfaust) [reply sklearnex/svm/_common.py]: Good Question! The changes made for LogisticRegression were made in 1.6, I'll change the comment not to mislead.

**Comment by icfaust** [comment]
> logreg changes look good to me. admittedly I do not know what is going on in the SVM part so i'll defer to others on review of that
> 
> There is one public CI fail related to **array** functionality which is modified in this code, but its also a test that fails often in public CI - do the error messages look real?

There is something odd with how it's install numpy, where its using a very old version with a broken array API support, which is messing with array_api_strict.  I'll give it a closer look.

**Comment by icfaust** [comment]
For reviewers: the last Nightly run on Main (September 24th 2025) https://dev.azure.com/daal/daal4py/_build/results?buildId=51485&view=results

**Comment by icfaust** [comment]
Failure in private CI due to an infrastructure issue.
